src/constructors/regular.c

```diff
@@ -311,12 +311,45 @@ int igraph_lattice(igraph_t *graph, const igraph_vector_t *dimvector,
 int igraph_ring(igraph_t *graph, igraph_integer_t n, igraph_bool_t directed,
                 igraph_bool_t mutual, igraph_bool_t circular) {
 
-    igraph_vector_t v = IGRAPH_VECTOR_NULL;
-
     if (n < 0) {
         IGRAPH_ERRORF("The number of vertices must be non-negative, got %" IGRAPH_PRId ".", n, IGRAPH_EINVAL);
     }
+    if(n == 1 && circular)
+    {
+        igraph_vector_t edges;
+        
+        //initialise edge vector and insert edge 0-0 for self loop
+        igraph_vector_init(&edges,2);
```

Comment:
You can use igraph_vector_init_int to clean things up.

Reply:
When you initialize a vector it needs to be added to the 'finally stack' with
IGRAPH_FINALLY(igraph_vector_destroy, &edges);

Reply:
Done